### DISCUSSION TITLE
Is it possible to use plugins which depend on other plugins?

### DISCUSSION THREAD

**Discussion by abhi18av** [nextflow-io/nextflow#1968] [Q&A] [answered]
Hi Paolo,

I was wondering that with the development of `plugins` in NF, would the roadmap include the possiblity of having different versions of same plugin (or Java lib) with the plugin/lib dependency tree. Is it similar to the way java handles it's deps?

For example

```
nf-plugin-1
  |
  |
  _ nf-amazon@1.0.0

nf-amazon@1.0.3
```

Or the above situation would never arise since all `nf plugins` are supposed to be designed in an independent manner instead of nesting.

**Comment by pditommaso** [comment, marked as answer]
Each plugin can depend on other plugins, but they should not have conflicting versions
